src/main/java/org/zalando/problem/spring/web/advice/validation/ConstraintViolationProblem.java

```diff
@@ -38,16 +40,16 @@
     public static final URI CONSTRAINT_VIOLATION = URI.create(CONSTRAINT_VIOLATION_VALUE);
 
     private final Optional<String> detail;
-    private final ImmutableList<Violation> violations;
+    private final List<Violation> violations;
 
-    public ConstraintViolationProblem(final ImmutableList<Violation> violations) {
-        this(Optional.empty(), violations);
+    public ConstraintViolationProblem(final List<Violation> violations) {
+        this(Optional.empty(), new ArrayList<>(violations));
     }
 
     @JsonCreator
-    private ConstraintViolationProblem(final Optional<String> detail, final ImmutableList<Violation> violations) {
```

Comment:
I guess you wanted to avoid the extra copying for internally created lists where we know that nobody is referencing them? Makes the code harder to read, imo. Especially since we know have two constructors that look similar but have differences in how they handle lists.